CRM/Member/ActionMapping.php

```diff
@@ -90,13 +90,26 @@ public function createQuery($schedule, $phase, $defaultParams) {
       $query['casDateField'] = 'e.' . $query['casDateField'];
     }
 
+    $recurStatuses = civicrm_api3('ContributionRecur', 'getoptions', [
+      'field' => "contribution_recur_contribution_status_id",
+    ])['values'] ?? [];
+    // Exclude the renewals that are cancelled or failed.
+    $nonRenewStatusIds = array_keys(array_intersect($recurStatuses, ['Cancelled', 'Failed']));
     // FIXME: Numbers should be constants.
     if (in_array(2, $selectedStatuses)) {
       //auto-renew memberships
-      $query->where("e.contribution_recur_id IS NOT NULL");
+      $query->join('cr', 'INNER JOIN civicrm_contribution_recur cr on e.contribution_recur_id = cr.id');
+      $query->where("e.contribution_recur_id IS NOT NULL AND cr.contribution_status_id NOT IN (#nonRenewStatusIds)")
```

Comment:
@jitendrapurohit is the NOT NULL aspect already covered in that your using an INNER rather than LEFT JOIN in L101?

Reply:
Thanks for spotting that @seamuslee001 . Fixed.